vnext/ReactWindowsCore/DevSettings.h

```diff
@@ -102,6 +102,10 @@ struct DevSettings {
   // Until the ABI story is addressed we'll use this instead of the above for
   // the purposes of selecting a JSI Runtime to use.
   JSIEngineOverride jsiEngineOverride{JSIEngineOverride::Default};
+
+  /// A directory which the executor may read or write to in order to perform
+  /// its needed functions (e.g. bytecode caching)
+  std::filesystem::path executorScratchDirectory;
```

Comment:
>path [](start = 19, length = 4)

We should be able to use C++ 17 everywhere now, so I went ahead and used this instead of passing strings around. Less consistent, but seems overall better.